script/dex-65062-PDF_Resubmit.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+class Recreate686PDFProcessor
+  def initialize(claim_ids, report_file)
+    @report_file = report_file
+    @service = BGS::Services.new(external_uid: '1234', external_key: 'tbtest')
+    @claims =  SavedClaim::DependencyClaim.where(id: claim_ids)
+    headers = %w[created_at zip_uploaded_at id file_number url]
+    csv = headers.to_csv
+    file = report_file
+    File.write(file, csv)
+  end
+
+  def get_form_hash_686c(bgs_person, form)
+    {
+      'veteran_information' => {
+        'full_name' => {
+          'first' => bgs_person[:first_nm],
+          'middle' => '',
+          'last' => bgs_person[:last_nm]
+        },
+        'common_name' => bgs_person[:first_nm],
+        'va_profile_email' => bgs_person[:email_addr],
+        'email' => form['dependents_application']['veteran_contact_information']['email'],
+        'participant_id' => bgs_person[:ptcpnt_id],
+        'ssn' => bgs_person[:ssn_nbr],
+        'va_file_number' => bgs_person[:file_nbr],
+        'birth_date' => form['dependents_application']['veteran_information']['birth_date'],
+        'uuid' => '',
+        'icn' => ''
+      }
+    }
+  end
+
+  def get_meta_data(doc_type, claim)
+    {
+      veteranFirstName: claim.parsed_form['veteran_information']['full_name']['first'],
+      veteranLastName: claim.parsed_form['veteran_information']['full_name']['last'],
+      fileNumber: claim.parsed_form['veteran_information']['va_file_number'],
+      source: 'BVA-VA.GOV resubmit',
+      docType: doc_type,
+      claimDate: claim['created_at'],
+      vaGovClaimId: claim['id']
+    }
+  end
+
+  def upload_pdf_submission_to_s3(claim:, doc_path:, return_url: false, url_life_length: 1.week.to_i,
+                                  doc_type: '686C-674')
+    metadata = get_meta_data(doc_type, claim)
+    zipname = "#{claim.id}.zip"
+    generate_zip_and_upload(zipname, metadata, return_url, url_life_length, doc_path)
+  end
+
+  def generate_zip_and_upload(zipname, metadata, return_url, url_life_length, doc_path)
+    zip_path_and_name = "tmp/#{zipname}"
+    Zip::File.open(zip_path_and_name, create: true) do |zipfile|
+      zipfile.get_output_stream('metadata.json') { |f| f.puts metadata.to_json }
+      doc_path.each do |file_path|
+        zipfile.add(file_path[9..], file_path)
+      end
+    end
+    s3_resource = Reports::Uploader.new_s3_resource
+    s3_bucket = Reports::Uploader.s3_bucket
+    obj = s3_resource.bucket(s3_bucket).object(zipname)
+    obj_ret = obj.upload_file(zip_path_and_name, content_type: 'application/zip')
+    if return_url
+      obj.presigned_url(:get, expires_in: url_life_length)
+    else
+      obj_ret
+    end
+  ensure
+    File.delete(zip_path_and_name)
+    doc_path.each do |file_path|
+      File.delete(file_path)
+    end
+  end
+
+  def process
+    @claims.each do |claim|
+      bgs_person = @service.people.find_by(ssn: claim.parsed_form['dependents_application']['veteran_information']['ssn'])
```

Comment:
You think you should put some error handling around here for instances where BGS errors or does not find an ID? 

For example wrap in begin/rescue and in the rescue write to the CSV like.... just the file number populated, and maybe another column for error_message or something?